ua/typereg.go

```diff
@@ -5,92 +5,87 @@
 package ua
 
 import (
-	"reflect"
 	"sync"
 
 	"github.com/gopcua/opcua/errors"
 )
 
-// TypeRegistry provides a registry for Go types.
-//
-// Each type is registered with a unique identifier
-// which cannot be changed for the lifetime of the component.
-//
-// Types can be registered multiple times under different
-// identifiers.
-//
-// The implementation is safe for concurrent use.
-type TypeRegistry struct {
-	mu    sync.Mutex
-	types map[string]reflect.Type
-	ids   map[reflect.Type]string
+type FuncRegistry struct {
```

Comment:
I think it deserves a comment but wrt to functions

Reply:
How do you feel about renaming the file from typereg.go -> funcreg.go ?